apps/asap-server/src/utils/sync-google-calendar.ts

```diff
@@ -0,0 +1,67 @@
+import { google, calendar_v3 as calendarV3, Auth } from 'googleapis';
+import logger from './logger';
+
+export type SyncCalendarFactory = (
+  syncToken: string | undefined,
+  syncEvent: (event: calendarV3.Schema$Event) => Promise<void>,
+  auth: Auth.GoogleAuth | Auth.OAuth2Client,
+) => (googleCalendarId: string) => Promise<string | undefined | null>;
+
+export const syncCalendarFactory: SyncCalendarFactory = (
+  syncToken: string | undefined,
+  syncEvent: (event: calendarV3.Schema$Event) => Promise<void>,
+  auth: Auth.GoogleAuth | Auth.OAuth2Client,
+) => {
+  const syncCalendar = async (googleCalendarId: string) =>
+    fetchEvents(googleCalendarId, auth, syncEvent, syncToken);
+
+  return syncCalendar;
+};
+
+const fetchEvents = async (
+  googleCalendarId: string,
+  auth: Auth.GoogleAuth | Auth.OAuth2Client,
+  syncEvent: (event: calendarV3.Schema$Event) => Promise<void>,
+  syncToken: string | undefined,
+  pageToken?: string,
+): Promise<string | undefined | null> => {
+  const params: calendarV3.Params$Resource$Events$List = {
+    pageToken: pageToken || undefined,
+    calendarId: googleCalendarId,
+    singleEvents: true, // recurring events come returned as single events
+  };
+
+  const calendar = google.calendar({ version: 'v3', auth });
+
+  const res = await calendar.events.list(params).catch((err) => {
```

Comment:
I would prefer not to mix async/await with promise chains. 

Since we are using async functions all over the place I would simply await the response and try/catch the error